### DISCUSSION TITLE
如何下一個好的prompt?

### DISCUSSION THREAD

**Discussion by Tchang323** [1101-GS4719-C/Python-Community-GS4719-C#2234] [General]
昨天上課聽了"AI agent and prompt engineering"的演講，受了蠻多的啟發，不管是生成式AI的能與不能、人機協作(AI agent)的工作型態，甚至式應用在業界的開發流程。而我也做了一些關於下prompt的筆記，大家也可以分享一些關於下prompt的撇步：
OpenAI 官方指南：
1. 撰寫清楚提示
    - 描述詳細訊息    誰是總統? vs. 誰是2021年墨西哥的總統?
    - 讓模型版演某角色
    - 使用分隔符號清楚地劃分不同區塊
2. 提供參考資訊
    - 指定輸出長度
    - 提供參考文本，在企業較常掛一個資料庫如mySQL給AI查(資料夠不夠好?很重要)
3. 將複雜任務拆解成子任務
    - 把子任務作分類，逐項定義操作需求
    - 總結或過濾之前的對話(模型不能記住所有過去對話紀錄)
    - 分段總結長文檔案
4. 給模型時間思考
    - 不要急著說出結論(你剛剛的回答對嗎?再想一遍/剛剛那三組答案你覺得哪一個最正確)
    - 詢問模型在思考過程中是否有遺漏內容
5. 使用外部工具
    - 外部API
    - 指定程式碼
6. 系統測試與優化
    - 人工評價
    - 電腦和人工綜合評價

**Comment by Chattenoire** [comment]
老實說，我覺得prompting有一個根本上的謬誤:
會想要使用AI獲得答案，其中一個大理由就是需要快速得到答案、沒時間自行蒐集資料，但是這些prompting technique本質上就是在叫人花更多時間；不管是few-shot/template或是其他奇奇怪怪的方法，這都違背上面的根本訴求。
